### PR TITLE
refgene2protein updates

### PR DESCRIPTION
**Pull request by mnedmonson**
- report transcript variant number
- also include NR_ records
- preprocessing option to speed parsing

### CODE REVIEW

File: src/scripts/refgene2protein.pl [hunk 1]
```diff
@@ -78,37 +97,114 @@ sub generate_refgene2protein {
...
+      my $fh_out = $wf->output_filehandle();
+
+      my $wf_chunk;
+      my $fh_chunk;
+      my $chunk_num = 0;
+      my $nm_count = 0;
+      my $chunk_size = 1000;
+
+      foreach my $infile (@infiles) {
+	printf STDERR "preprocessing %s...\n", $infile;
```
Comment (mcrusch) [inline src/scripts/refgene2protein.pl:121]: Looks like indentation gets messed up here.  Perhaps a mix of tabs and spaces?

File: src/scripts/refgene2protein.pl [hunk 2]
```diff
@@ -78,37 +97,114 @@ sub generate_refgene2protein {
...
-	}
+      $info->{protein} = "" unless exists $info->{protein};
+      # won't be present for NR_
+
+      my $acc = $info->{accession} || die;
+      my $protein = $info->{protein} || "";
+      # not present for NR_
+      my $usable = $acc =~ /^N[MR]_/ ? 1 : 0;
+      # exclude predicted XM_ records which don't appear in refFlat.txt
+      die "WTF" if $acc =~ /^NM_/ and !$protein;
```
Comment (mcrusch) [inline src/scripts/refgene2protein.pl:189]: More meaningful error message?

File: src/scripts/refgene2protein.pl (hunk 1 above)
Comment (mnedmonson) [reply src/scripts/refgene2protein.pl:121]: The indenting looks fine in my editor (emacs) which I believe uses a mix of both by default (tabs if the indent level is divisible by the tab length, which I believe is default 8).  I think it's possible to change the behavior to have every indent use a tab character but this looks much worse with e.g. the "less" pager.  Changing to that layout would I think also make every line in the file look different (is there some kind of "diff -w" option here?).

File: src/scripts/refgene2protein.pl (hunk 2 above)
Comment (mnedmonson) [reply src/scripts/refgene2protein.pl]: OK, updated

**Review by mcrusch — APPROVED** [review]

**Pull request merged**